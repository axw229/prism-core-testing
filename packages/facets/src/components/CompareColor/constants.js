export const baseClass = 'color__comparison'
export const containerClass = `${baseClass}__container`
export const containerHeaderClass = `${baseClass}__container__header`
export const containerHeaderButtonClass = `${baseClass}__container__header__button`
export const wrapperClass = `${baseClass}__wrapper`
export const prevBtnWrapperClass = `${baseClass}__prev-btn__wrapper`
export const buttonsClass = `${baseClass}__buttons`
export const buttonsVisibleClass = `${baseClass}__buttons--visible`
export const nextBtnWrapperClass = `${baseClass}__next-btn__wrapper`
export const backgroundColorClass = `${baseClass}__background__color`
export const queueWrapperClass = `${baseClass}__queue__wrapper`
