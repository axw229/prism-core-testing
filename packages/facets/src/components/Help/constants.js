export const baseClass = `cvw-help`
export const wrapper = `${baseClass}__wrapper`
export const tabsContainer = `${baseClass}__tabs-container`
export const contentWrapper = `${baseClass}__content-wrapper`
export const helpContent = `${baseClass}__help-content`
export const helpContentHide = `${helpContent}--hide`
export const contentHeader = `${baseClass}__content-header`
export const contentDetails = `${baseClass}__content-details`
export const subContent = `${baseClass}__subcontent`
export const subContentHeader = `${baseClass}__subContent-header`
export const subContentDetails = `${baseClass}__subContent-details`
export const helpIcons = `${baseClass}__help-icons`
export const helpImages = `${baseClass}__help-images`
export const helpImagesMobile = `${baseClass}__help-images-mobile`
export const iconWrap = `${baseClass}__icon-wrap`
export const iconWrapUndoRedo = `${baseClass}__icon-wrap-undoredo`
export const iconInfo = `${baseClass}__icon-info`
export const secondIcon = `${baseClass}__second-icon`
export const SCROLL_SPEED = 500
