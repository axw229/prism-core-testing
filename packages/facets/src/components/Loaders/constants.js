// @flow
// the 5 colors used by loaders in their default state; taken from SW CVW circa 2019
export const DEFAULT_COLORS = [
  '#00b9e4',
  '#69be28',
  '#ffb612',
  '#e17000',
  '#b71234'
]
// how many colors there are (this is sort of magic, so don't touch it)
export const NUM_COLORS = DEFAULT_COLORS.length
// how many generated colors are allowed to be dark
export const NUM_DARKS = Math.floor(NUM_COLORS / 2)
// how many generated colors should be light
export const NUM_LIGHTS = NUM_COLORS - NUM_DARKS
// default brightness step between generated colors
export const GENERATED_COLOR_BRIGHTNESS_STEP = 10
// % of brightness spectrum edges considered to be too dark or too light
export const GENERATED_COLOR_EDGE_PCT = 0.1
