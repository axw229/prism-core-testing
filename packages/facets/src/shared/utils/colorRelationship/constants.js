export const FULL = 360
export const HALF = FULL / 2
export const QUARTER = HALF / 2
export const NUM_CLUSTERS = 12
export const CLUSTER_SIZE = FULL / NUM_CLUSTERS
