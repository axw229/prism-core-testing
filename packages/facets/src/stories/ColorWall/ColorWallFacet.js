import { createPrismEmbedScript } from '../shared/utils'

export function createColorWallFacet ({ language, cdpPageRoot, prismVersion }) {
  const wrapper = document.createElement('div')
  const div = document.createElement('div')
  div.id = 'prism-embed-root'

  const script = createPrismEmbedScript((e) => void (0), prismVersion)
  div.setAttribute('prism-auto-embed', 'true')
  div.classList.add('prism-color-wall')
  div.setAttribute('data-prism-facet', 'ColorWallFacet')
  div.setAttribute('data-language', language)
  div.setAttribute('data-page-root', '/')
  div.setAttribute('data-color-detail-page-root',
    cdpPageRoot)
  div.setAttribute('data-route-type', 'hash')

  wrapper.appendChild(div)
  wrapper.appendChild(script)

  return wrapper
}
