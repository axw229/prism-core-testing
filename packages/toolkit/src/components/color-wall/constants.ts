export const MAX_SWATCH_SIZE = 50
export const BASE_SWATCH_SIZE = 15
export const OUTER_SPACING = 20
export const MIN_SCROLLER_HEIGHT = 200
export const MAX_SCROLLER_HEIGHT = MIN_SCROLLER_HEIGHT * 3

export const MIN_BASE_SIZE = 14
export const MAX_BASE_SIZE = 45
export const ZOOMED_BASE_SIZE = 50
export const ACTIVE_SWATCH_SIZE = 150
const BLOOMED_SWATCH_MIN_SIZE = 70
export const BLOOMED_SWATCH_SIZES = new Array(4).fill(null).map((v, i, arr) => {
  return BLOOMED_SWATCH_MIN_SIZE + ((ACTIVE_SWATCH_SIZE - BLOOMED_SWATCH_MIN_SIZE) / arr.length) * (arr.length - 1 - i)
})

export const SWATCH_WIDTH_WRAP_THRESHOLD = 20 // will prefer wrapped view if unwrapped view target swatch size falls below threshold value

export const TITLE_SIZE_RATIOS = {
  1: 0.4, // l1 should be styled regular weight
  2: 0.4, // l2 should be styled bold weight
  3: 0.5 // l3 should be styled bold weight, and is larger than l1 and l2
}

export const TITLE_SIZE_MIN = 12
export const TITLE_SIZE_MAX = 16
