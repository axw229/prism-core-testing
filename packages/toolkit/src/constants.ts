export enum SCENE_TYPES {
  OBJECT = 'objects',
  ROOM = 'rooms',
  AUTOMOTIVE = 'automotive',
  FAST_MASK = 'fastmask'
}

export enum DRAG_TYPES {
  SWATCH = 'SWATCH'
}

export enum SCENE_VARIANTS {
  DAY = 'day',
  NIGHT = 'night',
  MAIN = 'main'
}

// ERRORS
export const SYSTEM_ERROR = 'SYSTEM_ERROR'

export enum BUTTON_POSITIONS {
  TOP = 'top',
  BOTTOM = 'bottom'
}

export const KEY_CODES = {
  KEY_CODE_ARROW_DOWN: 40,
  KEY_CODE_ARROW_LEFT: 37,
  KEY_CODE_ARROW_RIGHT: 39,
  KEY_CODE_ARROW_UP: 38,
  KEY_CODE_ENTER: 13,
  KEY_CODE_SPACE: 32,
  KEY_CODE_TAB: 9
}
