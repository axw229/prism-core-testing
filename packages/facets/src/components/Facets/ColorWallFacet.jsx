// @flow
import React, { useCallback, useContext, useEffect, useMemo, useState } from 'react'
import { useIntl } from 'react-intl'
import { useDispatch, useSelector } from 'react-redux'
import { Route, Switch, useHistory } from 'react-router-dom'
import at from 'lodash/at'
import isArray from 'lodash/isArray'
import ColorWallContext, { colorWallContextDefault } from 'src/components/Facets/ColorWall/ColorWallContext'
import Search from 'src/components/Search/Search'
import SearchBar from 'src/components/Search/SearchBar'
import ConfigurationContext, {
  type ConfigurationContextType
} from 'src/contexts/ConfigurationContext/ConfigurationContext'
import facetBinder from 'src/facetSupport/facetBinder'
import { type FacetBinderMethods, facetBinderDefaultProps } from 'src/facetSupport/facetInstance'
import { type FacetPubSubMethods, facetPubSubDefaultProps } from 'src/facetSupport/facetPubSub'
import { fullColorName, generateColorWallPageUrl } from 'src/shared/helpers/ColorUtils'
import extendIfDefined from 'src/shared/helpers/extendIfDefined'
import useEffectAfterMount from 'src/shared/hooks/useEffectAfterMount'
import { type ColorsState } from 'src/shared/types/Actions.js.flow'
import translateBooleanFlexibly from 'src/shared/utils/translateBooleanFlexibly.util'
import { updateColorStatuses } from 'src/store/actions/loadColors'
import { setIsColorWallModallyPresented } from '../../store/actions/navigation'
import { GenericOverlay } from '../ToolkitComponents'
import ColorWall from './ColorWall/ColorWall'
import ColorWallRouter from './ColorWall/ColorWallRouter'
import ColorWallToolbar from './ColorWall/ColorWallToolbar/ColorWallToolbar'
import ColorWallV3 from './ColorWall/ColorWallV3'

type Props = FacetPubSubMethods &
  FacetBinderMethods & {
    addButtonText?: string,
    alwaysShowColorFamilies?: boolean,
    colorDetailPageRoot?: string,
    colorWallBgColor?: string,
    defaultSection?: string,
    displayAddButton?: boolean,
    displayInfoButton?: boolean,
    displayAddButtonText?: boolean,
    displayDetailsLink?: boolean,
    hiddenSections?: string | string[], // as string, "section name 1" or "section name 1|section name 2|etc" will be parsed into an array
    initialFocusId?: string | number
  }

export const EVENTS = {
  colorsLoaded: 'PRISM/out/colorsLoaded',
  decorateColors: 'PRISM/in/decorateColors',
  emitColor: 'PRISM/out/emitColor',
  selectedGroup: 'PRISM/out/selectedGroup',
  selectGroupAndColor: 'PRISM/in/selectGroupAndColor',
  selectGroup: 'PRISM/in/selectGroup',
  clearSection: 'PRISM/in/clearSection',
  loading: 'PRISM/in/loading',
  initialFocusId: 'PRISM/in/initialFocusId'
}

const SearchBarNoLabel = () => {
  const { messages = {} } = useIntl()
  const { uiStyle }: ConfigurationContextType = useContext(ConfigurationContext)

  return (
    <div className='color-wall-wrap__chunk'>
      <SearchBar
        className={uiStyle === 'minimal' ? 'SearchBarMinimal' : undefined}
        label={messages['SEARCH.FIND_A_COLOR']}
        placeholder={messages[uiStyle === 'minimal' ? 'SEARCH.SEARCH_FOR_A_COLOR' : 'SEARCH.SEARCH_BY']}
        showCancelButton
        showLabel={false}
      />
    </div>
  )
}

const SearchContain = () => <Search contain />

export const ColorWallPage = (props: Props) => {
  const {
    addButtonText,
    alwaysShowColorFamilies,
    colorDetailPageRoot,
    colorWallBgColor,
    initialFocusId,
    defaultSection,
    displayAddButton = false,
    displayAddButtonText,
    displayInfoButton = false,
    displayDetailsLink = true,
    hiddenSections,
    publish,
    subscribe,
    unsubscribeAll
  } = props
  const dispatch = useDispatch()
  const history = useHistory()
  const { cwv3 } = useSelector<ColorsState>((state) => state.colors)
  const { brandKeyNumberSeparator }: ConfigurationContextType = useContext(ConfigurationContext)

  // -----------------------------------------------------
  // accept and process color decoration from host
  useEffect(() => subscribe(EVENTS.decorateColors, handleColorDecoration), [])
  const handleColorDecoration = useCallback((decoratedColors) => dispatch(updateColorStatuses(decoratedColors)), [])
  const colorMap = useSelector((state) => at(state, 'colors.items.colorMap')[0])
  const { section, family } = useSelector((state) => at(state, 'colors')[0])
  useEffect(() => {
    colorMap && publish(EVENTS.colorsLoaded, colorMap)
  }, [colorMap])

  // -----------------------------------------------------
  // handle emitting selected color to host
  const emitColor = useSelector((state) => at(state, 'colors.emitColor')[0])
  // on color select AFTER initial mount
  useEffectAfterMount(() => {
    const color = emitColor && emitColor.color
    if (color) {
      // resetWall(true)
      publish(EVENTS.emitColor, color)
    }
  }, [emitColor && emitColor.timestamp])

  useEffect(() => {
    publish(EVENTS.selectedGroup, { family: family, section: section })
  }, [section, family])

  // -----------------------------------------------------
  // handle host demanding appearance of loading
  const [isLoading, updateLoading] = useState(false)
  useEffect(() => {
    subscribe(EVENTS.loading, updateLoading)
    return unsubscribeAll
  }, [])

  // -----------------------------------------------------
  // handle host changing section/family after initial load
  useEffect(() => {
    subscribe(EVENTS.selectGroup, ({ section, family }) => {
      history.push(generateColorWallPageUrl(section, family))
    })
    return unsubscribeAll
  }, [])

  // -----------------------------------------------------
  // handle host changing section/family/color after initial load
  useEffect(() => {
    subscribe(EVENTS.selectGroupAndColor, ({ section, family, color }) => {
      history.push(
        generateColorWallPageUrl(
          section,
          family,
          color.id,
          fullColorName(color.brandKey, color.colorNumber, color.name, brandKeyNumberSeparator)
        )
      )
    })
    return unsubscribeAll
  }, [])

  // -----------------------------------------------------
  // handle reseting section/family to defaults
  useEffect(() => {
    subscribe(EVENTS.clearSection, () => {
      history.push(generateColorWallPageUrl(defaultSection))
    })
  }, [])

  const [focusColorId, updateFocusColorId] = useState(!isNaN(initialFocusId) ? Number(initialFocusId) : undefined)
  useEffect(() => {
    subscribe(EVENTS.initialFocusId, (focusId) => {
      updateFocusColorId(!isNaN(focusId) ? Number(focusId) : undefined)
    })
  }, [])

  // -----------------------------------------------------
  // handle hidden sections
  const processedHiddenSections = useMemo(() => {
    if (typeof hiddenSections === 'string') {
      // hiddenSections is a single pipe-delimited string; break it into an array here before it gets into context
      return hiddenSections.split('|')
    } else if (isArray(hiddenSections)) {
      return hiddenSections
    }

    return []
  }, [hiddenSections])

  // -----------------------------------------------------
  // build color wall context and a11y state
  const cwContext = useMemo(
    () =>
      extendIfDefined({}, colorWallContextDefault, {
        addButtonText,
        colorDetailPageRoot,
        colorWallBgColor,
        initialFocusId: focusColorId,
        displayAddButton: translateBooleanFlexibly(displayAddButton),
        displayInfoButton: translateBooleanFlexibly(displayInfoButton),
        displayAddButtonText: translateBooleanFlexibly(displayAddButtonText),
        displayDetailsLink: translateBooleanFlexibly(displayDetailsLink),
        hiddenSections: processedHiddenSections
      }),
    [
      addButtonText,
      colorDetailPageRoot,
      colorWallBgColor,
      displayAddButton,
      displayAddButtonText,
      displayDetailsLink,
      focusColorId
    ]
  )

  // -----------------------------------------------------
  // handle unmounting
  useEffect(
    () => () => {
      // unsubscribe from everything on unmount
      dispatch(setIsColorWallModallyPresented())
      unsubscribeAll()
    },
    []
  )

  const CWToolbar = () => (
    <div className='color-wall-wrap__chunk'>
      <ColorWallToolbar alwaysShowColorFamilies={alwaysShowColorFamilies} />
    </div>
  )

  return (
    <ColorWallContext.Provider value={cwContext}>
      <ColorWallRouter defaultSection={defaultSection}>
        <div className='color-wall-wrap'>
          <nav>
            <Switch>
              <Route path='(.*)?/search/:query' component={SearchBarNoLabel} />
              <Route path='(.*)?/search' component={SearchBarNoLabel} />
              <Route path='(.*)?/section/:section/family/:family' component={CWToolbar} />
              <Route path='(.*)?/section/:section/family/' component={CWToolbar} />
              <Route path='(.*)?/family/:family/' component={CWToolbar} />
              <Route path='(.*)?/family/' component={CWToolbar} />
              <Route component={CWToolbar} />
            </Switch>
          </nav>
          <Switch>
            <Route path='(.*)?/search/:query' component={SearchContain} />
            <Route path='(.*)?/search/' component={SearchContain} />
            <Route component={cwv3 ? ColorWallV3 : ColorWall} />
          </Switch>
          {isLoading ? <GenericOverlay type={GenericOverlay.TYPES.LOADING} semitransparent /> : null}
        </div>
      </ColorWallRouter>
    </ColorWallContext.Provider>
  )
}

ColorWallPage.defaultProps = {
  ...facetPubSubDefaultProps,
  ...facetBinderDefaultProps,
  resetOnUnmount: true
}

export default facetBinder(ColorWallPage, 'ColorWallFacet')
