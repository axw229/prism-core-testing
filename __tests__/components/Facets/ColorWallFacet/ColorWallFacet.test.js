/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Route, Switch } from 'react-router-dom'
import { ColorWallFacet, RootRedirectColorWall, ColorWallComponent } from 'src/components/Facets/ColorWallFacet/ColorWallFacet'
import { ROUTE_PARAMS } from 'constants/globals'

const homeRoute = '/'
const colorWallBaseUrl = `/${ROUTE_PARAMS.ACTIVE}/${ROUTE_PARAMS.COLOR_WALL}`
const colorWallUrlPattern = `${colorWallBaseUrl}(/.*)?`
const locationProp = { pathname: homeRoute }

const getColorWallFacet = (props) => {
  let defaultProps = {
    location: locationProp
  }

  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<ColorWallFacet {...newProps} />)
}

describe('ColorWallFacet routes', () => {
  let colorWallFacet
  let pathMap = {}
  beforeAll(() => {
    if (!colorWallFacet) {
      colorWallFacet = getColorWallFacet()
    }

    pathMap = colorWallFacet.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props()
      pathMap[routeProps.path] = routeProps.component
      return pathMap
    }, {})
  })

  it('should match snapshot', () => {
    expect(colorWallBaseUrl).toMatchSnapshot()
  })

  it('should render Switch component', () => {
    expect(colorWallFacet.find(Switch).exists()).toBe(true)
  })

  it('should render Switch component with location prop defined as locationProp constant', () => {
    expect(colorWallFacet.find(Switch).prop('location')).toEqual(locationProp)
  })

  it('should have 2 Route components', () => {
    expect(colorWallFacet.find(Route)).toHaveLength(2)
  })

  it('should show RootRedirectColorWall component for route defined as homeRoute constant', () => {
    expect(pathMap[homeRoute]).toBe(RootRedirectColorWall)
  })

  it('should show ColorWallComponent component for route defined as colorWallUrlPattern constant', () => {
    expect(pathMap[colorWallUrlPattern]).toBe(ColorWallComponent)
  })
})
