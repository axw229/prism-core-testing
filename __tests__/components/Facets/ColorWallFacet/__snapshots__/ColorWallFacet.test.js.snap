// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorWallFacet routes should match snapshot 1`] = `"/active/color-wall"`;
